drivers/FlashIAP.cpp

```diff
@@ -134,6 +148,8 @@ int FlashIAP::program(const void *buffer, uint32_t addr, uint32_t size)
         addr += chunk;
         buf += chunk;
     }
+
+    mbed_mpu_manager_unlock_mem_xn();
```

Comment:
Do we need execute never lock/unlock to fetch the sector size also?

Reply:
The `get_*` functions should not need to execute from ram as they are just returning data from a table. If there is a case you are concerned about I can add it to these functions as well.